test/functional.jl

```diff
@@ -0,0 +1,108 @@
+module TestFunctional
+
+# nullable equality is a little strange... we need a "is a null with type T"
+# and a "is a nullable with value egal to k"
+
+# these functions are curried for readability
+isnull_typed(u::Nullable, T::Type) = typeof(u).parameters[1] == T && isnull(u)
+isnull_typed(t::Type) = u -> isnull_typed(u, t)
+
+isnullableof(u::Nullable, x) = !isnull(u) && u.value === x
+isnullableof(x) = u -> isnullableof(u, x)
+
+using Base.Test
+
+# getindex
+@test_throws NullException Nullable()[]
+@test_throws NullException Nullable{Int}()[]
+@test Nullable(0)[] == 0
+
+@test_throws BoundsError Nullable()[1]
```

Comment:
Also test a negative value, both on null and non-null. Test `Nullable()[2]`, and also `Nullable{Int}()`.